src/main/groovy/com/okta/test/mock/matchers/UrlMatcherTest.groovy

```diff
@@ -0,0 +1,69 @@
+package com.okta.test.mock.matchers
+
+import org.hamcrest.Matchers
+import org.hamcrest.collection.IsMapContaining
+import org.testng.annotations.Test
+
+import static com.okta.test.mock.matchers.UrlMatcher.singleQueryValue
+import static com.okta.test.mock.matchers.UrlMatcher.urlMatcher
+
+import static org.hamcrest.Matchers.*
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.text.MatchesPattern.matchesPattern
+
+class UrlMatcherTest {
+
+    @Test
+    void noQueryTest() {
+        assertThat "http://example.com/foo", urlMatcher("http://example.com/foo")
+    }
+
+    @Test
+    void emptyQueryTest() {
+        assertThat "http://example.com/foo?", urlMatcher("http://example.com/foo")
+    }
+
+    @Test
+    void singleValueQueryTest() {
+        assertThat "http://example.com/foo?one=two&three=four",
+                urlMatcher("http://example.com/foo",
+                    singleQueryValue("one", "two"))
```

Comment:
This is a bit difficult to read as is. Can we either:
- Break the `urlMatcher` params into newlines
- Bring `singleQueryValue` up

```java
assertThat "http://example.com/foo?one=two&three=four",
                urlMatcher("http://example.com/foo", singleQueryValue("one", "two"))
```